salt/cloud/clouds/azurearm.py

```diff
@@ -52,7 +52,7 @@
     from salt.utils.msazure import object_to_dict
     import azure.storage
     from azure.common.credentials import UserPassCredentials
-    from azure.mgmt.compute import (
+    from azure.mgmt.compute.compute_management_client import (
         ComputeManagementClient,
         ComputeManagementClientConfiguration,
```

Comment:
(remark: my alias is lmazuel, not imazuel :). Just to be sure to do not miss if you cite me)

Yes, some methods now return an `AzureOperationPoller` instance when the call in asynchronous. I saw you use it with `wait()` method which is good.
I found one you seems to have missed with `netconn.network_security_group.create_or_update`:
http://azure-sdk-for-python.readthedocs.io/en/latest/ref/azure.mgmt.network.operations.html#azure.mgmt.network.operations.NetworkSecurityGroupsOperations.create_or_update

This should allow you to avoid the `while True` loop. You can put your timeout inside the `wait` method:
http://msrestazure.readthedocs.io/en/msrestdoc/msrestazure.html#msrestazure.azure_operation.AzureOperationPoller.wait

Also, you can use `result` instead of `wait` to get at the same time the created object.